API-GUIDELINE.md

```diff
@@ -0,0 +1,254 @@
+#API GUIDELINES
+
+##Overview
+As commercetools aims to be 100% API First, developer experience for API solution becomes very important for us.
+Additionally we aim to achieve more automation and improve the review process of new changes in the APIs.
+
+With this in mind, we have created "API Guidelines" that aims to help developers to design APIs:
+
+ - simple
+ - consistent 
+ - self-explanatory
+ 
+The Keywords to have a good Guidelines are:
+ 
+ - Consistency => following the Guidelines
+ - Uniformity => checking in the code if there are other properties with similar name and adjust in base of it or improve it
+ - Explicability => explain with a few words the purpose of the property avoiding abbreviations (there are some *exceptions).
+
+##Scope
+The purpose of these guidelines is to achieve the definition of consistent practices and patterns for all API endpoints.
+
+##URI
+Our URIs are composed by:
+   
+    Base Uri like:
+        https://api.{region}.{cloudProvider}.commercetools.com 
```

Comment:
`api` is one service.
We also have `auth` for example.

Reply:
make sense I'm gonna specify that the URI sample is for the api service

Reply:
and for the other services there is the link below, what do you think about?

Reply:
which link? https://docs.commercetools.com/http-api.html#hosts ?
Have you seen https://docs.commercetools.com/http-api-authorization#requesting-an-access-token-using-commercetools-oauth2-server ?

Reply:
yes and yes ;) but I can mention the link of the authorization if this is what you mean

Reply:
My suggestion:
use `https://{service}.{region}.{cloudProvider}.commercetools.com `
and use the 2 links as examples.